app/page/send/index.tsx

```diff
@@ -150,12 +154,23 @@ export function Send() {
     return (
       <SelectAddress
         key="step2"
-        to={txTo}
-        setTokenSelected={setTokenSelected}
-        onChangeTo={setTxTo}
+        onStepBack={() => {
+          setTokenSelected(null)
+          navigate(Path.Send)
```

Comment:
nit: `navigate` could be removed like the select token PR.

Reply:
We need to navigate to `/send` because we will get `/send/${tokenAddress}` via clicking Token.
When we step back to SelectToken, it will parse the url. ([Ref](https://github.com/pilagod/waallet-extension/blob/82b93fec1984aa8817641bff0cfb191de43c29f7/app/page/send/index.tsx#L138-L143))

Reply:
I think if we re-order the code as [follows](https://github.com/pilagod/waallet-extension/blob/82b93fec1984aa8817641bff0cfb191de43c29f7/app/page/send/index.tsx#L138-L147), we might be able to remove the `navigate(Path.Send)`?

```ts
  if (!tokenSelected) {
    return <SelectToken key="step1" setTokenSelected={setTokenSelected} />
  }

  if (tokenSelected === null && params?.tokenAddress) {
    const token = tokens.find((token) => token.address === params.tokenAddress)
    if (token) {
      setTokenSelected(token)
    }
  }
```

Reply:
Um... If doing so, when we click token in home page, it will go to step1 (token selected) instead of parsing url param first and set `tokenSelected` 🧐

Reply:
Sorry, my mistake. The above code will cause `Send` page order errors from `Home - Token`.

@WIN-LIN  is probably right here, because if we remove `navigate(Path.Send)`, the Send page won't be able to go back to the previous `SelectToken` page when coming from `Home - Token`.

Reply:
I think I got the crux. We should use params to decide our initial token first, then later part of send page can be controlled purely by state.

nit: I found `useParams` could be handier than `useRoute`.

```tsx
  const params = useParams<{ tokenAddress?: string }>()
  // const [, params] = useRoute<{
  //   tokenAddress: string
  // }>(`${Path.Send}/:tokenAddress`)
  const tokens = getUserTokens()
  const initialToken = params.tokenAddress
    ? tokens.find((token) => token.address === params.tokenAddress)
    : null
  const [tokenSelected, setTokenSelected] =
    useState<Nullable<Token>>(initialToken)
  const [txTo, setTxTo] = useState<HexString>("")

  // if (tokenSelected === null && params?.tokenAddress) {
  //   const token = tokens.find((token) => token.address === params.tokenAddress)
  //   if (token) {
  //     setTokenSelected(token)
  //   }
  // }

```

Reply:
[apply useParams and remove navigate to Path.Send](https://github.com/pilagod/waallet-extension/pull/172/commits/c9a8a91acc13dce880ea060f98d69397c4217660)